usr/lib/thingy/thingy.py

```diff
@@ -189,6 +193,38 @@ def on_menu_quit(self, widget):
     def on_app_changed(self, widget):
         self.load_documents()
 
+    def on_search_changed(self, widget):
+        search_text = self.app_search.get_text().lower().strip()
+        self.filter_documents(filter=search_text)
+    
+    @_async
+    def filter_documents(self, filter=None):
+        
+        self.documents = []
+        self.clear_flowbox()
+
+        if(len(filter) == 0):
```

Comment:
I'd prefer using a different argument name than 'filter', since it's a python built-in function.

'text' or 'string' is fine.